### ISSUE TITLE
HNC: Investigate a test flake in hncconfig from `remove` to `propagate`

### ISSUE THREAD

**Issue by yiqigao217** [kubernetes-sigs/multi-tenancy#871]
Find the failed test logs [here](https://gist.github.com/yiqigao217/55c89db9c8e003067b39a2d011ab71c9).

It failed in #869  prow test on line https://github.com/kubernetes-sigs/multi-tenancy/blob/hnc-v0.4/incubator/hnc/internal/reconcilers/hnc_config_test.go#L361 .

I think it's irrelevant to #869 since it didn't change the current v1alpha1 API and v1alpha2 is not even served.

**Comment by yiqigao217** [comment]
This test is flaky in from `propagate` to `remove` too. See [failed test log](https://gist.github.com/yiqigao217/f51b0cf319839c4cf2e8871b094cfec2) in #866 .

**Comment by yiqigao217** [comment]
Other similar flaky tests:
#823 and [HNC #886 test flake logs](https://gist.github.com/yiqigao217/0861b03b0ee2d9cd900c82ccabbc61b5), [HNC #918 test flake logs](https://gist.github.com/yiqigao217/a297a48a8796f591c9c7fb9e36ecd546)

**Comment by yiqigao217** [comment]
Another one from #953 pre-submit - [log](https://gist.github.com/yiqigao217/02aa617bd7922533471c67badf079c79)
```
[Fail] HNCConfiguration [It] should unset MultipleConfigurationsForOneType if extra configurations are later removed 
```

**Comment by yiqigao217** [comment]
#990 pre-submit - [log](https://gist.github.com/yiqigao217/d6b1b21391c12c8de4bd452e48ea61bb)
```
[Fail] HNCConfiguration [It] should propagate objects if the mode of a type is changed from remove to propagate 
```

**Comment by GinnyJI** [comment]
This issue is addressed in #1003. If this issue does not happen again, then we can confirm that it's fixed and close this issue.

**Comment by adrianludwin** [comment]
/close

No need to wait IMO. If it's not fixed, we'll just open a new bug
